backend/wayland/backend.c

```diff
@@ -495,36 +486,32 @@ struct wlr_backend *wlr_wl_backend_create(struct wl_display *display,
 	if (remote_format == NULL) {
 		wlr_log(WLR_ERROR, "Remote compositor doesn't support DRM format "
 			"0x%"PRIX32, fmt);
-		goto error_renderer;
+		goto error_allocator;
 	}
 
 	const struct wlr_drm_format *render_format =
 		wlr_drm_format_set_get(render_formats, fmt);
 	if (render_format == NULL) {
 		wlr_log(WLR_ERROR, "Renderer doesn't support DRM format 0x%"PRIX32, fmt);
-		goto error_renderer;
+		goto error_allocator;
 	}
 
 	wl->format = wlr_drm_format_intersect(remote_format, render_format);
 	if (wl->format == NULL) {
 		wlr_log(WLR_ERROR, "Failed to intersect remote and render modifiers "
 			"for format 0x%"PRIX32, fmt);
-		goto error_renderer;
+		goto error_allocator;
 	}
 
 	wl->local_display_destroy.notify = handle_display_destroy;
 	wl_display_add_destroy_listener(display, &wl->local_display_destroy);
 
 	return &wl->backend;
 
-error_renderer:
-	wlr_renderer_destroy(wl->renderer);
 error_allocator:
 	wlr_allocator_destroy(wl->allocator);
-error_drm_fd:
-	close(wl->drm_fd);
-error_remote_display_src:
-	wl_event_source_remove(wl->remote_display_src);
```

Comment:
We need to remove the event source in the error path. Is there a reason why this line has been removed?